src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -533,7 +533,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     if (!$arg->getAttribute('tag')) {
                         throw new InvalidArgumentException(sprintf('Tag "<%s>" with type="tagged" has no or empty "tag" attribute in "%s".', $name, $file));
                     }
-                    $arguments[$key] = new TaggedIteratorArgument($arg->getAttribute('tag'));
+                    $arguments[$key] = new TaggedIteratorArgument($arg->getAttribute('tag'), $arg->getAttribute('index') ?: null);
```

Comment:
This doesn't respect the XML schema as index can only be an integer.
I don't think we can use the `key` attribute as it's used for array argument.
Do we need to introduce a new value to allow configuring the index attribute to use for tagged argument https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd#L226